src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/ThreadStatics.cs

```diff
@@ -17,35 +17,35 @@ namespace Internal.Runtime
     /// </summary>
     internal static class ThreadStatics
     {
+        [ThreadStatic]
+        private static object t_inlinedThreadStaticBase;
+
         /// <summary>
         /// This method is called from a ReadyToRun helper to get base address of thread
         /// static storage for the given type.
         /// </summary>
         internal static unsafe object GetThreadStaticBaseForType(TypeManagerSlot* pModuleData, int typeTlsIndex)
         {
-            if (typeTlsIndex >= 0)
-                return GetUninlinedThreadStaticBaseForType(pModuleData, typeTlsIndex);
-
-            ref object? threadStorage = ref RuntimeImports.RhGetInlinedThreadStaticStorage();
-            if (threadStorage != null)
-                return threadStorage;
+            if (typeTlsIndex < 0)
+                return t_inlinedThreadStaticBase;
 
-            return GetInlinedThreadStaticBaseSlow(ref threadStorage);
+            return GetUninlinedThreadStaticBaseForType(pModuleData, typeTlsIndex);
         }
 
-        [RuntimeExport("RhpGetInlinedThreadStaticBaseSlow")]
         internal static unsafe object GetInlinedThreadStaticBaseSlow(ref object? threadStorage)
         {
             Debug.Assert(threadStorage == null);
             // Allocate an object that will represent a memory block for all thread static fields
-            TypeManagerHandle typeManager = RuntimeImports.RhGetSingleTypeManager();
+            TypeManagerHandle typeManager = (new object()).GetMethodTable()->TypeManager;
```

Comment:
```suggestion
            TypeManagerHandle typeManager = EETypePtr.EETypePtrOf<object>().ToPointer()->TypeManager;
```
?

Reply:
(Also, I have noticed that we have unused broken `MethodTableOf<T>` - you can delete it while you are on it.)